packages/signers/src/lit-protocol/signer.ts

```diff
@@ -0,0 +1,197 @@
+    this._checkInternals();
+    return this._authContext as LitSessionSigsMap;
+  };
+
+  getAddress = async () => {
+    this._checkInternals();
+    const address = await this.signer?.getAddress();
+
+    return address as `0x${string}`;
+  };
+
+  signMessage = async (msg: Uint8Array | string) => {
+    this._checkInternals();
+
+    return this.signer?.signMessage(msg) as Promise<Address>;
+  };
+
+  signTypedData = (params: SignTypedDataParams) => {
+    this._checkInternals();
+
+    return this.signer?._signTypedData(
+      params.domain as TypedDataDomain,
+      params.types as any,
+      params.message
+    ) as Promise<Address>;
+  };
+
+  private _checkInternals() {
+    if (!this._authContext) {
+      throw new Error("Not Authenticated");
+    }
+
+    if (!this.signer) {
+      throw new Error("Signer is not initalized, did you call authenticate?");
+    }
+  }
+
+  private async _doAuthentication(props: LitAuthenticateProps<C>) {
+    // check if the object is structed as an auth method
+    // if so we sign the session key with the auth method
+    // as the auth material. If a session signature
+    // is provided then we skip this step.
+    if (Object.keys(props.context).indexOf("accessToken") > 0) {
+      const resourceAbilities = [
+        {
+          resource: new LitPKPResource("*"),
+          ability: LitAbility.PKPSigning,
+        },
+      ];
+      const sessionKeypair = props.sessionKeypair || generateSessionKeyPair();
+      const chain = props.chain || "ethereum";
+      let authNeededCallback: any;
+      if (props.context?.authMethodType === 1) {
+        authNeededCallback = async (params: AuthCallbackParams) => {
+          const response = await this.inner.signSessionKey({
+            statement: params.statement,
+            authMethods: [props.context as LitAuthMethod],
+            authSig: JSON.parse(props.context.accessToken as string) as AuthSig,
+            pkpPublicKey: `0x${this._pkpPublicKey}`,
+            expiration: params.expiration,
+            resources: params.resources,
+            chainId: 1,
+          });
+          return response.authSig;
+        };
+      } else {
+        authNeededCallback = async (params: AuthCallbackParams) => {
+          const response = await this.inner.signSessionKey({
+            statement: params.statement,
+            sessionKey: sessionKeypair,
+            authMethods: [props.context as LitAuthMethod],
+            pkpPublicKey: `0x${this._pkpPublicKey}`,
+            expiration: params.expiration,
+            resources: params.resources,
+            chainId: 1,
+          });
+          return response.authSig;
+        };
+      }
+
+      if (!this.inner.ready) {
+        await this.inner.connect();
+      }
+
+      const sessionSigs = await this.inner
+        .getSessionSigs({
+          chain,
+          expiration:
+            props.expiration ??
+            new Date(Date.now() + 1000 * 60 * 60 * 24 * 7).toISOString(),
```

Comment:
this being 1000 week in the future? is this arbitrary?

Reply:
yes our authentication steps check on expiration so we set the exp to a point well in the future.

Reply:
yes but this was a testing addition ill lower the default to `1 week`